src/main/kotlin/com/sourcegraph/cody/context/EmbeddingStatusView.kt

```diff
@@ -83,21 +80,23 @@ class EmbeddingStatusView(private val project: Project) : JPanel() {
   }
 
   private fun updateViewBasedOnStatus() {
-    val codebaseName = embeddingStatus.getMainText()
-    codebaseSelector.text = codebaseName.ifEmpty { "No repository" }
-    val icon = embeddingStatus.getIcon()
-    if (icon != null) {
-      embeddingStatusContent.icon = icon
-      embeddingStatusContent.preferredSize =
-          Dimension(icon.iconWidth + 10, embeddingStatusContent.height)
-    }
-    val tooltip = embeddingStatus.getTooltip(project)
-    if (tooltip.isNotEmpty()) {
-      embeddingStatusContent.setToolTipText(tooltip)
+    ApplicationManager.getApplication().invokeLater {
```

Comment:
I see all the logic is wrapped in `invokeLater`. Shouldn't be a caller's responsibility to call it on a proper thread? We could then mark it as `@RequiresEdt`.

Reply:
I have mixing feelings about that, let's discuss on the next standup.
I think this can go as it is for now.